src/compiler/scala/tools/nsc/backend/jvm/BTypes.scala

```diff
@@ -268,7 +260,7 @@ abstract class BTypes {
     def asPrimitiveBType : PrimitiveBType = this.asInstanceOf[PrimitiveBType]
   }
 
-  sealed trait PrimitiveBType extends BType {
+  sealed class PrimitiveBType(override val toString: String) extends BType {
```

Comment:
Could this be abstract?

Or it could be final and the instances vals

Reply:
Looks like its a sealed class for pattern matching over the cases. We don't have a way to do the same with val instances (although one could switch to Java enums for the same effect.)

Reply:
made abstract to make the matches exhaustive
@retronym WDYT about making the class final and making the matched properties in the rest of the class based on constructor params?

Reply:
It's use in patterns is pretty widespread so I would only do this if there is a some good motivating reason for the change.